src/org/jitsi/impl/neomedia/recording/SynchronizerImpl.java

```diff
@@ -324,21 +321,16 @@ private SSRCDesc getSSRCDesc(long ssrc)
      */
     private Endpoint getEndpoint(String endpointId)
     {
-        Endpoint endpoint = endpoints.get(endpointId);
-        if (endpoint == null)
+        synchronized (endpoints)
         {
-            synchronized (endpoints)
+            Endpoint endpoint = endpoints.get(endpointId);
```

Comment:
Yeah, I'm fine with those unprotected reads. Protecting the single read of "ssrcs" was a mistake.